src/CppUTest/MemoryLeakDetector.cpp

```diff
@@ -66,7 +66,7 @@ void SimpleStringBuffer::addMemoryDump(const void* memory, size_t memorySize)
 	size_t p;
 
     while (currentPos < memorySize) {
-        add("    %04lx: ", currentPos);
+        add("    %04hx: ", (unsigned short) currentPos);
```

Comment:
We can do that. I was assuming that if a memory dump was that big, it would probably be hard to glean the important information from anyway. I did find elsewhere that a selection based on sizeof() is pretty much the only sensible way to deal with these things (short of using features we are not supposed to use).